tracer/src/Datadog.Trace/Sampling/CustomSamplingRule.cs

```diff
@@ -76,17 +107,29 @@ public bool IsMatch(Span span)
                 return false;
             }
 
-            if (DoesNotMatch(input: span.ServiceName, pattern: _serviceNameRegex))
+            // special case where if both regex are null we match
+            if (_serviceNameRegex is null && _operationNameRegex is null)
             {
-                return false;
+                return true;
             }
 
-            if (DoesNotMatch(input: span.OperationName, pattern: _operationNameRegex))
+            try
+            {
+                // if the regex is null, we will always match it as we don't care
+                var serviceMatch = _serviceNameRegex?.Match(span.ServiceName).Success ?? true;
+                var operationMatch = _operationNameRegex?.Match(span.OperationName).Success ?? true;
+                return serviceMatch && operationMatch;
```

Comment:
This comment is super late, but if we don't match on service name then we can short-circuit and skip the operation name regex.

```csharp
return (_serviceNameRegex?.Match(span.ServiceName).Success ?? true) &&
       (_operationNameRegex?.Match(span.OperationName).Success ?? true);
```